src/util/BaseLoginRouter.js

```diff
@@ -231,6 +231,13 @@ function (Okta, BrowserFeatures, Settings,
           return;
         }
 
+        // Emit a beforeNavigate event with the current and next controller name
+        this.trigger(
+          'beforeNavigate',
```

Comment:
1. feels like the naming convention is not kind of consistent. ?

`navigated` / `beforeNavigate`
v.s.
`afterNavigate` / `beforeNavigate`

2. also `fromController` and `toController`?

Reply:
Yeah - Since `navigated` is essentially emitting "after navigation occurs", I wonder if it would make more sense to change these into:

- `beforeNavigate` -> `beforeNavigation`
- `controller` -> `fromController`

To me, this is much easier to understand:

```javascript
signIn.on('beforeNavigation', (data, context) => {
  // Assume view is transitioning between Primary Auth and MFA Push
  console.log(data);
  // { toController: 'mfa-verify' }
  console.log(context);
  // { fromController: 'primary-auth' }
});
```

Reply:
What I was thinking is the pattern `preRender` and `postRender` we have in Courage and iirc `before*` and `after*` is another sort of convention. (`beforeNavigation`, `afterNavigation`)
Even though `navigated` does emit "after navigation occurs", I feel with prefix it's easer to grasp.

Let's see what @rchild-okta thought?

Reply:
Is there a master list of the events we want to add for this first pass? It'd be easier to come up with the pattern we want with that in hand. In general, agree with @haishengwu-okta  that before/after (or some common prefixes, suffixes) is more flexible than depending on grammar for tenses. It'd be good to get this right up front - would be a good topic for the design meeting on Friday.